An option to use positive quadrature weights

fdkong opened the pull request:
Closes #19066

fdkong committed:
Added an option to force positive qweights if we need

fdkong committed:
Added a test using allow_negative_qweights

Closes #19066

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/845862/) on a1c8fab wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/19068/site)

This comment will be updated on new commits.

lindsayad reviewed (changes requested):
I didn't even know this flag existed in `QBase`

moosebuild commented:
Job [Generate and verify coverage](https://civet.inl.gov/job/845874/) on a1c8fab wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>2fa274</th>
      <th colspan=4>#19068 a1c8fa</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>81.64%</td>
      <td>81.64%</td>
      <td>+0.00%</td>
      <td>91.67%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>69807</td>
      <td>69821</td>
      <td>+14</td>
      <td>22</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15702</td>
      <td>15701</td>
      <td>-1</td>
      <td>2</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/19068/coverage_diff/framework.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/19068/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/19068/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/19068/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/19068/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/19068/coverage/contact)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/19068/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/19068/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/19068/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/19068/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/19068/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/19068/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/19068/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/19068/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/19068/coverage/navier_stokes)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/19068/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/19068/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/19068/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/19068/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/19068/coverage/rdg)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/19068/coverage/richards)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/19068/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/19068/coverage/tensor_mechanics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/19068/coverage/xfem)

</details>



This comment will be updated on new commits.

fdkong committed:
Apply suggestions from code review

const! haha

Co-authored-by: Alex Lindsay <alexlindsay239@gmail.com>

fdkong committed:
Updated docs for API createQRules in {Assembly.h}

fdkong committed:
Delete unneeded function

  void createQRules(QuadratureType type,
                    Order order,
                    Order volume_order,
                    Order face_order,
                    bool allow_negative_qweights = true);

fdkong committed:
Update framework/include/actions/SetupQuadratureAction.h

Co-authored-by: Guillaume Giudicelli <guillaume.giudicelli@gmail.com>

fdkong committed:
Pointed to the right md file SetupQuadratureAction.md

It automatically have a description for the parameter added
